packages/plugin-json/src/lib/amplience-schema-transformers.ts

```diff
@@ -158,10 +158,32 @@ export const objectProperties = (
             other: (type) =>
               ampliencePropertyType(prop, type, schema, schemaHost),
           }),
+          ...extensionProperty(prop, schema),
         },
       ]) ?? [],
   );
 
+const extensionProperty = (
+  prop: FieldDefinitionNode,
+  schema: GraphQLSchema,
+) => {
+  const extensionDirective = maybeDirective(prop, "amplienceExtension");
+  if (extensionDirective) {
+    const value = maybeDirectiveValue<StringValueNode>(
+      extensionDirective,
+      "name",
+    )?.value;
+    if (isObjectType(schema.getType(namedType(prop.type).name.value))) {
+      return {
+        "ui:extension": {
+          name: value,
+        },
+      };
+    }
+  }
+  return {};
```

Comment:
This function is not necessary, you can write this logic directly in the ampliencePropertyType function.